packages/@uppy/core/src/Uppy.ts

```diff
@@ -46,19 +51,86 @@ type FileRemoveReason = 'user' | 'cancel-all'
 
 type LogLevel = 'info' | 'warning' | 'error' | 'success'
 
-export type UnknownPlugin<M extends Meta, B extends Body> = InstanceType<
-  typeof BasePlugin<any, M, B> | typeof UIPlugin<any, M, B>
+export type UnknownPlugin<
+  M extends Meta,
+  B extends Body,
+  PluginState extends Record<string, unknown> = Record<string, unknown>,
+> = InstanceType<
+  | typeof BasePlugin<any, M, B, PluginState>
+  | typeof UIPlugin<any, M, B, PluginState>
 >
```

Comment:
@Murderlon I feel like I'm missing some TS nuance here. Wouldn't this be equivalent to
```suggestion
> = BasePlugin<any, M, B, PluginState>
```
If not, what is the difference?